src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp

```diff
@@ -229,6 +248,24 @@ class boss_sapphiron : public CreatureScript
                 return 0;
             }
 
+            ObjectGuid GetGUID(int32 data) const override
+            {
+                if (data == DATA_BLIZZARD_TARGET)
+                {
+                    // Filtering blizzards from summon list
+                    std::vector<Unit*> blizzards;
+                    for (ObjectGuid summonGuid : summons)
+                        if (Unit* temp = ObjectAccessor::GetUnit(*me, summonGuid))
+                            if (temp->GetEntry() == NPC_BLIZZARD)
```

Comment:
You can check creature entry on the guid first

Reply:
how? o.o